app/workers/send_user_invitation_worker.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class SendUserInvitationWorker < ApplicationJob
+  queue_as :default
+
+  RETRY_ERRORS = [
+    OpenSSL::SSL::SSLError,
+    Net::SMTPAuthenticationError,
+    Net::SMTPFatalError,
+    Net::SMTPServerBusy,
+    Net::SMTPSyntaxError,
+    Net::SMTPUnknownError,
+    Net::SMTPUnsupportedCommand,
+    SocketError,
+  ].freeze
+
+  DISCARD_ERRORS = [
+    ActiveRecord::RecordNotFound
+  ].freeze
+
+  def perform(invitation_id)
+    invitation = Invitation.find(invitation_id)
```

Comment:
This is ok, but IMO it is better enqueuing the `invitation` object directly instead of the ID, so internally it does it by [GlobalID](https://edgeguides.rubyonrails.org/active_job_basics.html#globalid).
And in the rescue error, instead of `ActiveRecord::RecordNotFound`, it is `ActiveJob::DeserializationError` is the object is not in the DB anymore.

Reply:
I don't have a strong opinion about it 🤷‍♂️

Reply:
It is the same but with less code because GlobalID would do the (de)serialization for us...

Doing `SendUserInvitationWorker.perform_later(self)` inside the invitation, and here just doing `def perform(invitation)` does not mean saving the entire object in Redis and then fetching it all here... it internally saves the object class and the ID only, and to deserialize, it does the `invitation = Invitation.find(invitation_id)` internally.

Please read this:

- [ActiveJob GlobalID](https://edgeguides.rubyonrails.org/active_job_basics.html#globalid)
- [Usage of GlobalID](https://github.com/rails/globalid#usage) in general. So you can see a bit of what the suggested code would do internally.